traffic_ops/traffic_ops_golang/riak.go

```diff
@@ -0,0 +1,204 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/apache/incubator-trafficcontrol/lib/go-log"
+	"github.com/apache/incubator-trafficcontrol/traffic_ops/tostructs"
+	"github.com/basho/riak-go-client"
+	"github.com/jmoiron/sqlx"
+	"github.com/lestrrat/go-jwx/jwk"
+	"io/ioutil"
+	"net/http"
+)
+
+const riak_port = 8087
+const cdn_uri_keys_bucket = "cdn_uri_sig_keys" // riak namespace for cdn uri signing keys.
+
+func getStringValue(resp *riak.FetchValueResponse) (string, error) {
+	var obj *riak.Object
+
+	if len(resp.Values) > 0 {
+		obj = resp.Values[0]
+	} else {
+		return "", fmt.Errorf("no such object")
+	}
+	return string(obj.Value), nil
+}
+
+func assignDeliveryServiceUriKeysKeysHandler(db *sqlx.DB, cfg Config) AuthRegexHandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request, p PathParams, username string, privLevel int) {
+		handleErr := func(err error, status int) {
+			log.Errorf("%v %v\n", r.RemoteAddr, err)
+			w.WriteHeader(status)
+			fmt.Fprintf(w, http.StatusText(status))
+		}
+		defer r.Body.Close()
+
+		xmlId := p["xml-id"]
+		data, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		// validate that the received data is a valid jwk keyset
+		var keys jwk.SymmetricKey
```

Comment:
This is incorrect. You'll need a custom type that probably looks something like:

    type URISignerKeyset struct {
        jwk.Set
        RenewalKid string `json:"renewal_kid,omitempty"`
    }
    type URISignerConfig map[string]URISignerKeyset

Reply:
Here's the readme: https://gist.github.com/alficles/a375f64b563993b608e69ec5a75a13be

(I tossed the README in a gist since the actual plugin isn't quite polished for submission to upstream yet.)

Reply:
And an extra thing you could check, if you wanted: Verify that ∃ exactly one `renewal_kid` and that it matches a key present in the file. Also, that every key has a `kid`. (It's legal to have JWKs without them, but we need them for efficient key-selection.)

Reply:
Okay, my understanding was that we would get a JSON web key set, RFC 7517, from NBC.  So, I can make the change with a custom struct and check the details.  So, I presume that NBC will now what to send us right?